esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -324,3 +324,251 @@ def __call__(self, cube):
         return _regrid_unstructured_to_rectilinear__perform(
             cube, regrid_info, self.mdtol
         )
+
+
+def _regrid_along_grid_dims(regridder, data, grid_x_dim, grid_y_dim, mdtol):
+    # The mesh will be assigned to the first dimension associated with the
```

Comment:
So this comment describes what happens in the edge case where there is a dimension between the latitude and longitude dimensions. I don't think we have any reason to expect this in real data so I don't know if there's an established convention for this, but we still have to decide how to handle these cases generally, so this comment acts as a precise description of behaviour. This behaviour is also described in the docstring of `regrid_rectilinear_to_unstructured`, but I have repeated the description here since this is where that behaviour derives from.